app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java

```diff
@@ -125,5 +129,49 @@ public void add(Message message) {
         return null;
     }
 
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+        switch (getItemViewType(position)) {
+            case VIEWTYPE_MESSAGE_HEADER:
+                final MessageHeaderParent messageHeaderParent = (MessageHeaderParent) getItem(position);
+                final MessageHeaderParent.MessageHeaderHolder messageHeaderHolder = ((MessageHeaderParent.MessageHeaderHolder) holder);
+
+                if (messageHeaderParent.getMessageType() == MessageType.STREAM_MESSAGE) {
+                    messageHeaderHolder.streamTV.setText(messageHeaderParent.getStream());
+                    messageHeaderHolder.topicTV.setText(messageHeaderParent.getSubject());
+
+                    ViewCompat.setBackgroundTintList(messageHeaderHolder.arrowHead, ColorStateList.valueOf(messageHeaderParent.getColor()));
+                    messageHeaderHolder.streamTV.setBackgroundColor(messageHeaderParent.getColor());
+
+                    if (messageHeaderParent.isMute())
+                        messageHeaderHolder.muteMessageImage.setVisibility(View.VISIBLE);
+
+                } else { //PRIVATE MESSAGE
+                    messageHeaderHolder.streamTV.setText(privateHuddleText);
+                    messageHeaderHolder.streamTV.setTextColor(Color.WHITE);
+                    messageHeaderHolder.topicTV.setText(messageHeaderParent.getDisplayRecipent());
+                    ViewCompat.setBackgroundTintList(messageHeaderHolder.arrowHead, ColorStateList.valueOf(mDefaultStreamHeaderColor));
+                    messageHeaderHolder.streamTV.setBackgroundColor(mDefaultStreamHeaderColor);
+                }
+                break;
+            case VIEWTYPE_MESSAGE:
+
+                MessageHolder messageHolder = ((MessageHolder) holder);
+                final Message message = ((Message) items.get(position));
+                messageHolder.contentView.setText(message.getFormattedContent(zulipApp));
+
+                if (message.getType() == MessageType.STREAM_MESSAGE) {
+                    messageHolder.senderName.setText(message.getSender().getName());
+                    messageHolder.leftBar.setBackgroundColor(message.getStream().getColor());
+                    messageHolder.messageTile.setBackgroundColor(Color.WHITE);
+                } else {
+                    messageHolder.senderName.setText(message.getSender().getName());
+                    messageHolder.leftBar.setBackgroundColor(mDefaultPrivateMessageColor);
+                    messageHolder.messageTile.setBackgroundColor(mDefaultPrivateMessageColor);
+                }
+                break;
+        }
```

Comment:
default case won't be needed because in `getItemViewType(int position)` function we are throwing an exception if the viewType is not known!
https://github.com/kunall17/zulip-android/blob/patch-9-recyclerview/app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java#L181-L183